src/main/java/com/blackducksoftware/integration/hub/nexus/event/handler/NexusTaskEventStoppedCanceledHandler.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * hub-nexus
+ *
+ * 	Copyright (C) 2018 Black Duck Software, Inc.
+ * 	http://www.blackducksoftware.com/
+ *
+ * 	Licensed to the Apache Software Foundation (ASF) under one
+ * 	or more contributor license agreements. See the NOTICE file
+ * 	distributed with this work for additional information
+ * 	regarding copyright ownership. The ASF licenses this file
+ * 	to you under the Apache License, Version 2.0 (the
+ * 	"License"); you may not use this file except in compliance
+ * 	with the License. You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * 	Unless required by applicable law or agreed to in writing,
+ * 	software distributed under the License is distributed on an
+ * 	"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * 	KIND, either express or implied. See the License for the
+ * 	specific language governing permissions and limitations
+ * 	under the License.
+ */
+package com.blackducksoftware.integration.hub.nexus.event.handler;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.sonatype.nexus.events.EventSubscriber;
+import org.sonatype.nexus.scheduling.NexusTask;
+import org.sonatype.nexus.scheduling.events.NexusTaskEventStoppedCanceled;
+import org.sonatype.sisu.goodies.common.ComponentSupport;
+
+import com.blackducksoftware.integration.hub.nexus.repository.task.AbstractHubWalkerTask;
+import com.blackducksoftware.integration.hub.nexus.repository.task.PolicyCheckTask;
+import com.blackducksoftware.integration.hub.nexus.repository.task.ScanTask;
+import com.blackducksoftware.integration.hub.nexus.util.ParallelEventProcessor;
+import com.google.common.eventbus.Subscribe;
+
+@Named
+@Singleton
+public class NexusTaskEventStoppedCanceledHandler extends ComponentSupport implements EventSubscriber {
+    private final ParallelEventProcessor parallelEventProcessor;
+
+    @Inject
+    public NexusTaskEventStoppedCanceledHandler(final ParallelEventProcessor parallelEventProcessor) {
+        this.parallelEventProcessor = parallelEventProcessor;
+    }
+
+    @Subscribe
+    public void handle(final NexusTaskEventStoppedCanceled<Object> nexusTaskEventStoppedCanceled) {
+        if (isBlackduckTask(nexusTaskEventStoppedCanceled.getNexusTask())) {
+            createLogger().info("Cancelling blackduck task safely");
+            parallelEventProcessor.hardShutdown();
+        }
+    }
+
+    private boolean isBlackduckTask(final NexusTask<?> task) {
+        final List<Class<? extends AbstractHubWalkerTask>> taskNames = Arrays.asList(ScanTask.class, PolicyCheckTask.class);
```

Comment:
Let's change this to a set.  The contains method in a set will be more efficient than iterating through the list of class names.  Create a HashSet of the class names.  The contains method will use the hashcode method to determine if the class is in the set or not.  More efficient than iterating through the list potentially.